lib/theme_javascript_compiler.rb

```diff
@@ -0,0 +1,213 @@
+class ThemeJavascriptCompiler
+
+  module PrecompilerExtension
+    def initialize(theme_id)
+      super()
+      @theme_id = theme_id
+    end
+
+    def discourse_node_manipulator
+      <<~JS
+
+      // Helper to replace old themeSetting syntax
+      function generateHelper(settingParts) {
+        const settingName = settingParts.join('.');
+        return {
+            "path": {
+              "type": "PathExpression",
+              "original": "theme-setting",
+              "this": false,
+              "data": false,
+              "parts": [
+                "theme-setting"
+              ],
+              "depth":0
+            },
+            "params": [
+              {
+                type: "NumberLiteral",
+                value: #{@theme_id},
+                original: #{@theme_id}
+              },
+              {
+                "type": "StringLiteral",
+                "value": settingName,
+                "original": settingName,
+              }
+            ],
+            "hash": {
+              "type": "Hash",
+              "pairs": [],
+            }
+        }
+      }
+
+      function manipulatePath(path){
+        // Override old themeSetting syntax when it's a param inside another node
+        if(path.parts[0] == "themeSetting"){
```

Comment:
Trivial, but I imagine this code isn't linted? Seems well tested I just noticed the spacing is off :)